app/components/Navigation.tsx

```diff
@@ -0,0 +1,101 @@
+import { memo, useState, Fragment, useMemo } from "react";
+import Home from "~/components/Home";
+import Profile from "~/components/Profile";
+import { useAuth0 } from "@auth0/auth0-react";
+import { Menu, Transition } from '@headlessui/react'
+
+//TODO: fix style of profile picture (maybe expand the image when the menu is opened as well)
+
+const Navigation = memo(function Nav({ apiBaseURL }: { apiBaseURL: string }){
+    const [darkMode, setDarkMode] = useState(true);
+    const [pageChanged, setPageChanged] = useState(true);
+
+    const changePage = (item: number) => {
+        switch(item){
+            case 1:
+                setPageChanged(true);
+                break;
+            case 2:
+                setPageChanged(false);
+                break;
+            case 3:
+                setDarkMode(darkMode => !darkMode);
+                break;  
+            case 4:
+                isAuthenticated ? logout() : loginWithRedirect();
+                break;       
+        }
+    }
+
+    const { loginWithRedirect, logout, user, isAuthenticated, isLoading } = useAuth0();
+
+    const userProfile = {
+        name: user?.name ?? "John Doe",
+        email: user?.email ?? 'johndoe@example.com',
+        imageUrl: isAuthenticated ? user?.picture : 'https://www.vecteezy.com/vector-art/550980-user-icon-vector',
+    }
+
+    const userNavigation = [
+        { name: 'Home', page: 1 },
+        { name: 'Profile', page: 2 },
+        { name: darkMode ? 'Light Mode' : 'Dark Mode', page: 3},
+        { name: isAuthenticated ? 'Sign in' : 'Sign out', page: 4},
+    ]
+      
+    function classNames(...classes: any[]) {
+        return classes.filter(Boolean).join(' ')
+    }
+
+    return (
+        <>  
+        <div className={darkMode ? "dark dark:bg-slate-800" : "bg-slate-100"}>
+            <nav className="sticky top-0 z-50 w-full px-5 py-2 flex justify-between items-center bg-slate-100 dark:bg-slate-800 border-b border-gray-300 dark:border-gray-600">
+                <div className="px-2 nav-align">
+                    <div className="px-1">
+                        <h1 className='text-5xl font-medium text-black dark:text-white p-3 pt-3' style={{left: "0%"}}> { pageChanged ? "Alignment Feed" : "Profile" } </h1>
```

Comment:
Might it be worth passing in the title as a parameter to the component, so it doesn't need to hardcode every known page title here and keep a boolean of the page it is currently on?